.changelog/3982.txt

```diff
@@ -0,0 +1,11 @@
+```release-note:enhancement
+resource/mongodbatlas_search_index: Adds the `num_partitions` attribute 
```

Comment:
```suggestion
resource/mongodbatlas_search_index: Adds `num_partitions` attribute 
```
nit: Just to stay consistent with other entries (applies to the 3 entries) https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/changelog-process.md#resource-and-provider-enhancements